client/components/categories/singleCategory.js

```diff
@@ -1,27 +1,39 @@
 import React, { Component } from 'react'
 import { connect } from 'react-redux'
 import { getCategoriesThunk } from '../../store/actionCreators'
+import { getProductsThunk } from '../../store/productThunks'
+import ProductCard from '../productCard'
 
 
 
 class SingleCategory extends Component {
 
   render() {
-    const { match: { params: { name } } } = this.props
+    console.log(this.props)
+    const { match: { params: { name, id } }, categories, products } = this.props
+    console.log(categories, products)
     return (
       <>
-        <h1>{name}</h1>
+        <h1>{name.toUpperCase()}</h1><br />
+        <div>
+          {products.filter(product => product.categoryId === id).map(cProduct => (
+            <div key={cProduct.id}>
+              <ProductCard product={cProduct} />
+            </div>
+          ))}
+        </div>
       </>
     )
   }
 }
 
-const mapStateToProps = (state) => ({
-  categories: state.categories
-})
+const mapStateToProps = (state) => {
+  return { categories: state.categories, products: state.products }
+};
 
 const mapDispatchToProps = (dispatch) => ({
```

Comment:
are you planning to use these functions later? i don't see any reference to them in your component.

Reply:
Hey, I thought they might be needed later since there is the issue that the products are not loading. I can remove them though.